be/src/exec/cross_join_node.cpp

```diff
@@ -691,4 +700,15 @@ pipeline::OpFactories CrossJoinNode::decompose_to_pipeline(pipeline::PipelineBui
     return left_ops;
 }
 
+pipeline::OpFactories CrossJoinNode::decompose_to_pipeline(pipeline::PipelineBuilderContext* context) {
+    using namespace pipeline;
+
+    if (runtime_state()->enable_spill() && _join_op == TJoinOp::CROSS_JOIN) {
+        return _decompose_to_pipeline<SpillableNLJoinBuildOperatorFactory, SpillableNLJoinProbeOperatorFactory>(
```

Comment:
would the join op be other types?

Reply:
The latter may be supported. But for now it looks like cross join support is enough